src/EditorFeatures/CSharpTest/GoToNextAndPreviousMember/CSharpGoToNextAndPreviousMemberTests.cs

```diff
@@ -0,0 +1,571 @@
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.GoToNextAndPreviousMember)]
+        [WorkItem(4311, "https://github.com/dotnet/roslyn/issues/4311")]
+        public void NextFromBetweenMethodsInTrailingTrivia()
+        {
+            var code = @"
+class C
+{
+    void M1()
+    {
+    } $$
+
+    [||]void M2() { }
+}";
+
+            AssertNavigated(code, next: true);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.GoToNextAndPreviousMember)]
+        [WorkItem(4311, "https://github.com/dotnet/roslyn/issues/4311")]
+        public void PreviousFromBetweenMethodsInTrailingTrivia()
+        {
+            var code = @"
+class C
+{
+    [||]void M1()
+    {
+    } $$
+
+    void M2() { }
+}";
+
+            AssertNavigated(code, next: false);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.GoToNextAndPreviousMember)]
+        [WorkItem(4311, "https://github.com/dotnet/roslyn/issues/4311")]
+        public void StopsAtExpressionBodiedMember()
+        {
+            var code = @"
+class C
+{
+    int M1() => $$42;
+
+    [||]int M2() => 42;
+}";
+
+            AssertNavigated(code, next: true);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.GoToNextAndPreviousMember)]
+        [WorkItem(4311, "https://github.com/dotnet/roslyn/issues/4311")]
+        public void NextInScript()
+        {
+            var code = @"
+$$void M1() { }
+
+[||]void M2() { }";
+
+            AssertNavigated(code, next: true, sourceCodeKind: SourceCodeKind.Script);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.GoToNextAndPreviousMember)]
+        [WorkItem(4311, "https://github.com/dotnet/roslyn/issues/4311")]
+        public void PrevInScript()
+        {
+            var code = @"
+[||]void M1() { }
+
+$$void M2() { }";
+
+            AssertNavigated(code, next: false, sourceCodeKind: SourceCodeKind.Script);
+        }
+
+        private static void AssertNavigated(string code, bool next, SourceCodeKind? sourceCodeKind = null)
+        {
+            var kinds = sourceCodeKind != null
+                ? SpecializedCollections.SingletonEnumerable(sourceCodeKind.Value)
+                : new[] { SourceCodeKind.Regular, SourceCodeKind.Script };
+            foreach (var kind in kinds)
+            {
+                using (var workspace = TestWorkspaceFactory.CreateWorkspaceFromLines(
+                    LanguageNames.CSharp,
+                    new CSharpCompilationOptions(OutputKind.DynamicallyLinkedLibrary),
+                    CSharpParseOptions.Default.WithKind(kind),
+                    code))
+                {
+                    var hostDocument = workspace.DocumentWithCursor;
+                    var document = workspace.CurrentSolution.GetDocument(hostDocument.Id);
+                    Assert.Empty(document.GetSyntaxTreeAsync().Result.GetDiagnostics());
+                    var targetPosition = GoToNextAndPreviousMethodCommandHandler.GetTargetPosition(
```

Comment:
The only cases are those two where GetCommandState returns unavailable.  I'm not sure the test setup effort is worth it here.